### PR TITLE
[Pruning] Basic pruning blocks storage & database upgrade

### PR DESCRIPTION
**Pull request by SergiySW**
Splitting https://github.com/nanocurrency/nano-node/pull/2881

### CODE REVIEW

File: nano/secure/ledger.cpp [hunk 1]
```diff
@@ -787,6 +785,9 @@ void nano::ledger::initialize (nano::generate_cache const & generate_cache_a)
 			this->cache.cemented_count += cemented_count_l;
 		});
 	}
+
+	auto transaction (store.tx_begin_read ());
+	cache.pruned_count = store.pruned_count (transaction);
```
Comment (guilhermelawless) [inline nano/secure/ledger.cpp:790]: Cache initialization isn't tested, could add it to `ledger.cache` test maybe?

Comment (SergiySW) [reply nano/secure/ledger.cpp:790]: Updated with test

File: nano/core_test/block_store.cpp [hunk 2]
```diff
@@ -1232,6 +1270,81 @@ TEST (block_store, online_weight)
...
+		// Add one
+		store->pruned_put (transaction, hash1);
+		ASSERT_TRUE (store->pruned_exists (transaction, hash1));
+	}
+
+	// Confirm that it can be found
+	{
+		auto transaction (store->tx_begin_read ());
+		ASSERT_EQ (store->pruned_count (transaction), 1);
+	}
```
Comment (guilhermelawless) [inline nano/core_test/block_store.cpp]: These single-action transactions can be made into one-liners
`ASSERT_EQ (store->pruned_count (store->tx_begin_read ()), 1);`

Comment (SergiySW) [reply nano/core_test/block_store.cpp]: Updated

**Review by guilhermelawless — DISMISSED** [review]

**Review by wezrule — DISMISSED** [review]

**Review by guilhermelawless — APPROVED** [review]

**Review by wezrule — APPROVED** [review]

**Pull request merged**